$NetBSD: patch-CMakeLists.txt,v 1.3 2021/07/21 12:53:20 yhardy Exp $

Rename VERSION to ARX_VERSION so that __has_include(<version>) does not
pick it up on a case-insensitive file system on Darwin.

--- CMakeLists.txt.orig	2021-07-14 00:04:33.000000000 +0000
+++ CMakeLists.txt
@@ -274,11 +274,11 @@ include(CheckIncludeFiles)
 include(CheckSymbolExists)
 include(CheckTypeSize)
 
-set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake") # For custom cmake modules
+set (CMAKE_MODULE_PATH "/tmp/pkgsrc/games/arx-libertatis/work/.buildlink/cmake-Modules" "/tmp/pkgsrc/games/arx-libertatis/work/.buildlink/cmake-Modules" "/tmp/pkgsrc/games/arx-libertatis/work/.buildlink/cmake-Modules" "${PROJECT_SOURCE_DIR}/cmake") # For custom cmake modules
 
 include(VersionString)
 set(version_names VERSION ICON_NAME DATA_VERSION IO_LIBRARY_ABI_VERSION)
-parse_version_file("${version_names}" "VERSION")
+parse_version_file("${version_names}" "ARX_VERSION")
 if(VERSION_ERROR OR DATA_VERSION_ERROR OR IO_LIBRARY_ABI_VERSION_ERROR)
 	message(WARNING "Error parsing version file!")
 endif()
@@ -1481,7 +1481,7 @@ configure_file("src/${PLATFORM_CONFIG_H}
 
 set(VERSION_TEMPLATE "src/core/Version.cpp.in")
 set(VERSION_FILE     "${PROJECT_BINARY_DIR}/core/Version.cpp")
-set(VERSION_SOURCES  VERSION "VERSION" AUTHORS "AUTHORS" COPYING "COPYING")
+set(VERSION_SOURCES  VERSION "ARX_VERSION" AUTHORS "AUTHORS" COPYING "COPYING")
 file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/core")
 version_file("${VERSION_TEMPLATE}" "${VERSION_FILE}" "${VERSION_SOURCES}" ".git")
 list(APPEND ARX_SOURCES ${VERSION_FILE})
@@ -1490,7 +1490,7 @@ if(WIN32)
 	set(VERSION_RC_TEMPLATE "src/core/Version.rc.in")
 	function(create_version_resource exe name)
 		set(rc_file "${PROJECT_BINARY_DIR}/${exe}-version.rc")
-		set(rc_sources VERSION "VERSION" COPYING "COPYING")
+		set(rc_sources VERSION "ARX_VERSION" COPYING "COPYING")
 		if(ARGC GREATER 2)
 			list(APPEND rc_sources ${ARGV2} "${ARGV2}")
 		endif()
@@ -1502,7 +1502,7 @@ if(WIN32)
 	set(APPLICATION_MANIFEST_TEMPLATE "src/core/Version.manifest.in")
 	function(create_application_manifest exe)
 		set(manifest_file "${PROJECT_BINARY_DIR}/${exe}.manifest")
-		set(manifest_sources VERSION "VERSION")
+		set(manifest_sources VERSION "ARX_VERSION")
 		set(manifest_defines "-DEXE=${exe}")
 		if(ARGC GREATER 1 AND ARGV1 STREQUAL "dpiaware")
 			list(APPEND manifest_defines "-DDPIAWARE=true")
